packages/generate/src/syntax/toEdgeQL.ts

```diff
@@ -223,22 +220,31 @@ export function $toEdgeQL(this: any) {
           continue;
         }
 
-        // set withBlock to top-level parent scope
-        const resolvedParentScopes = parentScopes.map(
-          parentScope => withVars.get(parentScope)?.scope ?? parentScope
+        const [firstParentScopeChain, ...parentScopeChains] = parentScopes.map(
```

Comment:
Took me a few read throughs to grok what was happening here, but in my attempt to form a coherent question, I was able to figure it out 😆